npm/vite-dev-server/cypress.config.ts

```diff
@@ -2,11 +2,17 @@ import { defineConfig } from 'cypress'
 import { devServer } from './dist'
 
 export default defineConfig({
-  'pluginsFile': 'cypress/plugins.js',
   'video': false,
   'fixturesFolder': false,
   'component': {
     'supportFile': 'cypress/support.js',
+    setupNodeEvents (_on, config) {
+      if (config.env.RUN_SIGNATURE) {
+        require('./cypress/new-signature/plugins.js')
```

Comment:
I believe this file should contain nothing since it is not containing the dev-server anymore.
If it does contain stuff, it should somehow trigger the function.

```suggestion
        require('./cypress/new-signature/plugins.js')(_on, config)
```

I would prefer a more ESM approach though:
Import at the top, use in the function.

Reply:
So I'm looking at this,

1 - what is the purpose of the two test scripts in the vite-dev-sever? In develop, the default configuration does not setup a devServer to run all of the tests. The new-signature-plugin setups up a dev-server and scopes the tests to the smoke test. It seems in 10.0-release, all tests are running with the devServer configuration setup. Should the original default be e2e tests? or can these be merged into one suite? It appears test are passing fine, but  I don't understand the original test intent to insure we aren't missing something.
- https://github.com/cypress-io/cypress/blob/develop/npm/vite-dev-server/package.json#L10-L11
- https://github.com/cypress-io/cypress/blob/develop/npm/vite-dev-server/cypress/new-signature/plugins.js 

2- it seems I've uncovered a bug 😢 when I made the changes to be:
```diff
   // package.json
-   "cy:run-signature": "RUN_SIGNATURE=true yarn cy:run",
+   "cy:run-signature": "yarn cy:run --env RUN_SIGNATURE=true",
   
   // cypress.config.ts
    setupNodeEvents (_on, config) {
      if (config.env.RUN_SIGNATURE) {
-        require('./cypress/new-signature/plugins.js')
+        config.specPattern = '**/smoke.cy.ts'
      }

      return config
    },
```
the updated specPattern is not being honored/used in the tests. //cc @tgriesser

Reply:
The purpose of these tests are:
1. check that vite-dev-server works with the `startDevServer` function for backward compatibility
2. check that it works with the new `devServer` signature

This may be broken and not work now.
I can check